opencti-platform/opencti-front/src/private/components/techniques/attack_patterns/AttackPatternsMatrixColumns.tsx

```diff
@@ -0,0 +1,276 @@
+  };
+
+  const toggleMode = () => handleToggleModeOnlyActive || setModeOnlyActive((prev) => !prev);
+  const onKillChainChange = (e: React.ChangeEvent<{ value: string }>) => {
+    setCurrentKillChain(e.target.value);
+  };
+
+  useEffect(() => {
+    const subscription = MESSAGING$.toggleNav.subscribe({
+      next: () => setNavOpen(localStorage.getItem('navOpen') === 'true'),
+    });
+    return () => subscription.unsubscribe();
+  }, []);
+
+  // Get deduplicated & sorted kill chains
+  const killChains = useMemo(
+    () => Array.from(
+      new Set(
+        attackPatternsMatrix?.attackPatternsOfPhases
+          ?.map((a) => a.kill_chain_name),
+      ),
+    ).sort((a, b) => a.localeCompare(b)),
+    [attackPatternsMatrix],
+  );
+
+  // Handle kill chain changes
+  useEffect(() => {
+    if (killChains.length > 0) {
+      const initialKillChain = selectedKillChain && killChains.includes(selectedKillChain)
+        ? selectedKillChain
+        : killChains[0];
+      setCurrentKillChain(initialKillChain);
+    }
+  }, [killChains, selectedKillChain]);
+
+  const getLevel = (ap: AttackPattern): number => {
+    const matchCount = attackPatterns.filter((n) => n.id === ap.attack_pattern_id || (ap.subAttackPatternsIds?.includes(n.id))).length;
+    const maxCount = Math.max(...attackPatterns.map((n) => {
+      const all = [n, ...(n.parentAttackPatterns?.edges || []).map((e) => e.node)];
+      return all.length;
+    }));
+    return computeLevel(matchCount, 0, maxCount, 0, 10);
+  };
+
+  const filteredData = useMemo(() => attackPatternsMatrix?.attackPatternsOfPhases
+    ?.filter((a) => a.kill_chain_name === currentKillChain)
+    .sort((a, b) => a.x_opencti_order - b.x_opencti_order)
+    .map((a) => ({
+      ...a,
+      attackPatterns: a.attackPatterns
+        ?.filter((ap) => !searchTerm
+        || ap.name.toLowerCase().includes(searchTerm.toLowerCase())
+        || ap.description?.toLowerCase().includes(searchTerm.toLowerCase())
+        || ap.x_mitre_id?.toLowerCase().includes(searchTerm.toLowerCase())
+        || ap.subAttackPatternsSearchText?.toLowerCase().includes(searchTerm.toLowerCase()))
+        .map((ap) => ({
+          ...ap,
+          id: ap.attack_pattern_id,
+          entity_type: 'Attack-Pattern',
+          level: getLevel(ap),
+        }))
+        .filter((o) => (modeOnlyActive ? o.level > 0 : o.level >= 0))
+        .sort((f, s) => f.name.localeCompare(s.name)),
+    })), [attackPatternsMatrix, currentKillChain, searchTerm, modeOnlyActive, attackPatterns]);
+
+  const matrixWidth = useMemo(() => {
+    if (marginRight) {
+      return navOpen ? 'calc(100vw - 430px)' : 'calc(100vw - 305px)';
+    }
+    return navOpen ? 'calc(100vw - 235px)' : 'calc(100vw - 110px)';
+  }, [marginRight, navOpen]);
+
+  return (
+    <UserContext.Consumer>
+      {({ bannerSettings }) => {
+        const matrixHeight = 310 + (bannerSettings?.bannerHeightNumber || 0) * 2;
```

Comment:
Try to put values like 430, 305, 235, 110, 310 into const that we can name and understand. We hardly know how the matrixWidth is calculated with this configuration.